internal/newrelic/client.go

```diff
@@ -97,18 +100,38 @@ func (nrc *nrClient) FindEntityMetrics(sample, customTagKey, entityTag string) (
 	return resultMetrics(a.Results), nil
 }
 
-func (nrc *nrClient) NRQLQuery(query, customTagKey, entityTag string) error {
+func (nrc *nrClient) NRQLQuery(query, customTagKey, entityTag string, errorExpected bool, expectedResults []spec.TestNRQLExpectedResult) error {
 	query = fmt.Sprintf("%s WHERE %s = '%s'", query, customTagKey, entityTag)
 
 	a, err := nrc.client.Query(nrc.accountID, query)
 	if err != nil {
+		if errorExpected {
+			return nil
```

Comment:
This isn't going to work, it appears that the way this function is expected to run is if there is an errorExpected, let the error return and then validate _from the caller_ that the error was indeed expected, and return the error w/ the fact that an error was expected, not to replace the functionality of this function entirely. for example: https://github.com/newrelic/newrelic-integration-e2e-action/blob/main/internal/runtime/nrql_tester.go#L26-L40 

from an engineering perspective, we might be expecting an error from the query itself, but we probably aren't expecting other types of errors, e.g. 'this error failed to run because the request failed to contact the server'. we don't want to cover up errors of unexpected nature, e.g. networking in the given example, which this rewrite will do, so this is not the best approach. 

this function, as named, should only really run NRQL queries, and anything extra, such as comparing for the expected values and checking whether or not an error was expected, should be handled by the caller of this function, rather than within this specific function itself. 
i would recommend you either have the caller handle the cases you want, or, create a specific function that runs nrql queries and compares them to the expected results.